public/app/types/accessControl.ts

```diff
@@ -50,6 +52,10 @@ export enum AccessControlAction {
   ActionServerStatsRead = 'server.stats:read',
 }
 
+export type ObjectWithMetadata = {
```

Comment:
Instead of doing this would it be a good idea to introduce an interface that you can extend in the `DataSourceSettings` interface?

Reply:
Sounds better to me as well. So I'd define something like this instead:
```tsx
export interface WithAccessControlMetadata {
  accessControl?: Record<string, boolean>;
};
```
I feel like I should move this definition to grafana-data instead, right?